client/packages/common/src/intl/currency/currency.ts

```diff
@@ -2,16 +2,27 @@ import currency from 'currency.js';
 import { useAuthContext } from '../../authentication';
 import { useIntlUtils } from '../utils';
 
+// See: https://stackoverflow.com/questions/50673859/how-to-prevent-very-small-numbers-from-being-converted-to-scientific-notation
+// For example 4e-7 becomes "0.0000004" instead of "4e-7"
+const toNonScientificString = (value: number): string => {
+  const valueString = value.toString();
+  const decimalsPart = valueString.split('.')?.[1] ?? '';
+  const eDecimals = Number(valueString?.split('e-')?.[1]) || 0;
+  const countOfDecimals = decimalsPart.length + eDecimals;
+  return value.toFixed(countOfDecimals);
+};
```

Comment:
Can't you do this with Intl.NumberFormat? 

```js
const n = 0.0000004
console.log(n) // 4e-7 
const f = new Intl.NumberFormat("en", {maximumFractionDigits: 21}).format(n)
console.log(f) // "0.0000004"
```

I've been trying to make all number formatting localisable as much as possible e.g. the NumericTextInput, so we want to avoid assuming what the decimal character should be. (So you pass the current locale into Intl.NumberFormat)